fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/properties/DerivationsProperty.java

```diff
@@ -839,6 +840,12 @@ public Derivations visitSortPlan(@Nonnull final RecordQuerySortPlan sortPlan) {
             return derivationsFromSingleChild(sortPlan);
         }
 
+        @Nonnull
+        @Override
+        public Derivations visitRecursivePlan(@Nonnull final RecordQueryRecursivePlan recursivePlan) {
+            return Derivations.EMPTY; // todo https://github.com/FoundationDB/fdb-record-layer/issues/2974
```

Comment:
Can't this just use whatever the level based-plan does? With the other properties I am fine not doing it right now especially if there are already issues filed, but this one is dangerous. Especially, since derivations from below are not propagated at all.

Reply:
We do not have (yet) an implementation of the derivations for recursive union, there is an open GitHub issue about this which I referenced also here, considering that, as you said, solving the issue will end up with an implementation for a recursive union plan that can be, for the most part, reused for recursive plan as well.

The issue is https://github.com/FoundationDB/fdb-record-layer/issues/2974.

Reply:
I updated derivation calculation for `RecordQueryRecursiveLevelUnionPlan` and `visitRecursiveDfsJoinPlan` after an offline discussion such that they return a combined derivation of their children. As it turned out this is not entirely correct, but definitely better than returning nothing. Therefore, the bug #2974 is still outstanding and this is by no means a proper fix.